app/celery/tasks.py

```diff
@@ -287,58 +297,57 @@ def save_smss(self, service_id: Optional[str], signed_notifications: List[Any],
         send_notification_to_queue(
-            notification,
-            research_mode,
+            notification_obj,
+            service.research_mode,
             queue=queue,
         )
 
         current_app.logger.debug(
             "SMS {} created at {} for job {}".format(
-                notification.id,
-                notification.created_at,
-                notification.job,
+                notification_obj.id,
+                notification_obj.created_at,
+                notification_obj.job,
             )
         )
 
 
 @notify_celery.task(bind=True, name="save-emails", max_retries=5, default_retry_delay=300)
 @statsd(namespace="tasks")
-def save_emails(self, service_id: Optional[str], signed_notifications: List[Any], receipt: Optional[UUID]):
+def save_emails(self, _service_id: Optional[str], signed_notifications: List[Any], receipt: Optional[UUID]):
     """
     Function that takes a list of signed notifications, stores
     them in the DB and then sends these to the queue. If the receipt
     is not None then it is passed to the RedisQueue to let it know it
     can delete the inflight notifications.
     """
-    verified_notifications: List[Any] = []
+    verified_notifications: List[VerifiedNotification] = []
     notification_id_queue: Dict = {}
-    saved_notifications = []
+    saved_notifications: List[Notification] = []
     for signed_notification in signed_notifications:
         try:
-            notification = signer.verify(signed_notification)
+            _notification: NotificationDictToSign = signer.verify(signed_notification)
         except BadSignature:
             current_app.logger.exception(f"Invalid signature for signed_notification {signed_notification}")
             raise
-        service_id = notification.get("service_id", service_id)  # take it it out of the notification if it's there
+        service_id = _notification.get("service_id", _service_id)  # take it it out of the notification if it's there
         service = dao_fetch_service_by_id(service_id, use_cache=True)
         template = dao_get_template_by_id(
-            notification.get("template"), version=notification.get("template_version"), use_cache=True
+            _notification.get("template"), version=_notification.get("template_version"), use_cache=True
         )
-        sender_id = notification.get("sender_id")
-        notification_id = notification.get("id", create_uuid())
-        notification["notification_id"] = notification_id
+        # bug: _notification does not have key "sender_id"
+        sender_id = _notification.get("sender_id")  # type: ignore
+        notification_id = _notification.get("id", create_uuid())
         reply_to_text = ""  # type: ignore
-
         if (
-            "reply_to_text" in notification and notification["reply_to_text"]
+            "reply_to_text" in _notification and _notification["reply_to_text"]
```

Comment:
are you renaming to indicate private variables? by default everything is private within the scope of the func?

Reply:
So before, we just had one `notification` variable that started out containing the signed data, and then the code added additional keys one by one. I wanted to define what keys are present in the signed data (that's the `NotificationDictToSign` type), and what additional keys need to be there before the notification is created in the db (that's the `VerifiedNotification` type).  

So that meant having two variables instead of one:  `_notification` for the signed data and `notification` for the signed data plus the additional keys we need to add in.

But I could probably have chosen different names that are a bit more descriptive.